Src/FluentAssertions/Events/EventHandlerFactory.cs

```diff
@@ -13,7 +13,7 @@ internal static class EventHandlerFactory
         /// Generates an eventhandler for an event of type eventSignature that calls RegisterEvent on recorder
         /// when invoked.
         /// </summary>
-        public static Delegate GenerateHandler(Type eventSignature, IEventRecorder recorder)
+        public static Delegate GenerateHandler(Type eventSignature, IEventRecording recorder)
```

Comment:
It seems this can only ever by an `EventRecorder`.
I haven't experimented, but wouldn't this crash if we try to call `EventRecorder.RecordEvent` on a `IEventRecording` not being instance of `EventRecorder`?

Reply:
It will throw a NRE.